src/executables/sysrepo-plugind.c

```diff
@@ -631,6 +616,20 @@ main(int argc, char **argv)
         goto cleanup;
     }
 
+    /* create notification rotation change subscription */
+    if ((r = sr_module_change_subscribe(sess, "sysrepo-plugind", "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/enabled",
+            sr_my_notif_cb, &opts, 0, SR_SUBSCR_ENABLED | SR_SUBSCR_DONE_ONLY, &subscr))) {
+        error_print(r, "Failed to subscribe");
+        goto cleanup;
+    }
+
+    /* create notification rotation state data change subscription */
+    if ((r = sr_oper_get_subscribe(sess, "sysrepo-plugind", "/sysrepo-plugind:sysrepo-plugind/notif-datastore/rotation/archived-files-count",
+            sr_my_state_cb, &opts, 0, &subscr))) {
+        error_print(r, "Failed to subscribe");
+        goto cleanup;
+    }
+
```

Comment:
`sysrepo-plugin`'s purpose is to load *plugins* from a well-known path, and start executing these. Now, using `sysrepo-plugind` as an executor for this "notification aging" makes sense, but it should be a proper plugin, not something hardcoded in the executable's `main()`.

Reply:
It seemed redundant to make it into a plugin but your are right that making it an "internal" plugin (that would not be dynamically loaded) would be a cleaner solution.